File: InteractiveLive/Android/VideoOne/solutions/interactivelive/src/main/java/com/vertcdemo/solution/interactivelive/view/TopFadeRecyclerView.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.solution.interactivelive.view;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

public class TopFadeRecyclerView extends RecyclerView {

    public TopFadeRecyclerView(@NonNull Context context) {
        super(context);
    }

    public TopFadeRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public TopFadeRecyclerView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected float getBottomFadingEdgeStrength() {
        return 0;
    }
}

File: Client/Android/solutions/vod-live-feed/src/main/java/com/byteplus/vodlive/view/TopFadeRecyclerView.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.byteplus.vodlive.view

import android.content.Context
import android.util.AttributeSet
import androidx.recyclerview.widget.RecyclerView

class TopFadeRecyclerView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : RecyclerView(context, attrs, defStyleAttr) {
    override fun getBottomFadingEdgeStrength() = 0F
}